### ISSUE TITLE
[469a] DM-FNB-Spaceport map freezing and/or crashing during match

### ISSUE THREAD

**Issue by TankBeef** [OldUnreal/UnrealTournamentPatches#200]
Hello. Just like the title says, it happens with any renderer, full screen or windowed, most often with bots and at least 5 minutes of game. Have to CTRL ALT Delete to exit. Also happenes with the DOM version.

**Comment by OjitroC** [comment]
Which audio driver are you using? Have you tried the map with a different audio driver?

**Comment by TankBeef** [comment]
> Which audio driver are you using? Have you tried the map with a different audio driver?

OpenAL. It is the only one that works for me. Cluster crashes some maps, and Galaxy sounds awful. I do not feel that it has to do with audio though. I was thinking maybe some conflict with the XPickups, but then the other maps work for me. Well, a similar crash happens with the Orion's Curse assault map. But the other FNB maps I have played do not give me issues.

**Comment by TankBeef** [comment]
> Which audio driver are you using? Have you tried the map with a different audio driver?

Oh, I also tested with DM-Refine, the map has the same umx. No crash.

**Comment by OjitroC** [comment]
The reason I asked about audio drivers is because someone on UT99.org had a problem with FNB's UnrealGateways when using Cluster and switching to OpenAL solved it.  I also thought XPickups in relation to that crash as that's something Spaceport and UnrealGateways have in common but if other maps with XPickups work Ok then it's unlikely to be that.

There's no inherent problems with either of those maps as they run OK in 436 and I can run both in 469b with Galaxy (the only audio driver I can use).  

So there's nothing in the log to indicate the reason for the freeze/crash?

**Comment by TankBeef** [comment]
> The reason I asked about audio drivers is because someone on UT99.org had a problem with FNB's UnrealGateways when using Cluster and switching to OpenAL solved it. I also thought XPickups in relation to that crash as that's something Spaceport and UnrealGateways have in common but if other maps with XPickups work Ok then it's unlikely to be that.
> 
> There's no inherent problems with either of those maps as they run OK in 436 and I can run both in 469b with Galaxy (the only audio driver I can use).
> 
> So there's nothing in the log to indicate the reason for the freeze/crash?

Unreal Gateways never crashes on me. I actually wrote to Rob FNB before posting here, he told me the maps work best with 436 and OpenGL. I will test tonight again and post the log.

**Comment by TankBeef** [comment]
Ok, I switched to Galaxy (I don't understand why it gives me a very distorted sound, most noticeable with the pulse gun. But that's another issue.) Tested again, froze at aprox. 8 minutes of play, with 16 bots. This is what the log shows me:

Log: Log file open, 02/03/21 20:13:08
Init: Name subsystem initialized
Init: Version: 469
Init: Compiled: Sep 22 2020 12:41:55
Init: Command line: 
Init: Base directory: C:\Unreal Tournament\System\
Init: Character set: Unicode
Log: Bound to Engine.dll
Log: Bound to Core.dll
Log: Bound to Window.dll
Init: Object subsystem initialized
Init: Computer: DESKTOPR3NEK3D
Init: User: herre
Init: Memory total: Phys=4194303K Pagef=4194303K Virt=4194176K
Init: Working set: 32000 / 159000
Init: Timer Frequency=2904.003245 MHz - Timing Source = RDTSC
Init: CPU Page size=4096, Processors=4
Init: CPU Detected: PentiumPro-class processor (GenuineIntel)
Init: CPU Features: CMov FPU RDTSC PAE MMXSSE KNI
Init: Unreal engine initialized
Log: Bound to WinDrv.dll
Init: Enhanced Pointer Precision Enabled - Mouse info: 0 0 0
Init: Initializing DirectDraw
Init: DirectDraw drivers detected:
Init:    display (Primary Display Driver)
Init: DirectDraw: Enumerated Display Modes:
Init: DirectDraw initialized successfully
Init: Registering Mouse for RawInput.
Init: RawInput Devices registered.
Init: Client initialized
Log: Bound to Render.dll
Init: Lighting subsystem initialized
Init: Rendering initialized
Log: LoadMap: Entry
Log: Bound to Fire.dll
Log: Bound to IpDrv.dll
Log: Game class is 'UTIntro'
Log: Bringing Level Entry.MyLevel up for play (200)...
ScriptLog: InitGame: 
ScriptLog: Base Mutator is Entry.Mutator0
Log: Browse: UT-Logo-Map.unr?Name=TankBeef?Class=BotPack.TMale2?team=255?skin=SoldierSkins_HOFjester.Word?Face=SoldierSkins_HOFjester.Tommy?Voice=BotPack.VoiceMaleTwo?OverrideClass=
Log: LoadMap: UT-Logo-Map.unr?Name=TankBeef?Class=BotPack.TMale2?team=255?skin=SoldierSkins_HOFjester.Word?Face=SoldierSkins_HOFjester.Tommy?Voice=BotPack.VoiceMaleTwo?OverrideClass=
Log: Collecting garbage
Log: Purging garbage
Log: Unloading: Package WinDrv
Log: Unloading: Package Render
Log: Garbage: objects: 15776->15774; refs: 177602
Log: Game class is 'UTIntro'
Log: Bringing Level UT-Logo-Map.MyLevel up for play (200)...
ScriptLog: InitGame: ?Name=TankBeef?Class=BotPack.TMale2?team=255?skin=SoldierSkins_HOFjester.Word?Face=SoldierSkins_HOFjester.Tommy?Voice=BotPack.VoiceMaleTwo?OverrideClass=
ScriptLog: Base Mutator is UT-Logo-Map.Mutator0
Init: Initialized moving brush tracker for Level UT-Logo-Map.MyLevel
Log: Bound to UWeb.dll
ScriptLog: Team 255
ScriptLog: Login: TankBeef
Log: Possessed PlayerPawn: TMale2 UT-Logo-Map.TMale0
Init: Input system initialized for WindowsViewport0
Log: Opened viewport
Log: Bound to OpenGLDrv.dll
Log: Initializing OpenGLDrv...
Log: Enter SetRes()
Init: Registering Mouse for RawInput.
Init: Using RIDEV_CAPTUREMOUSE
Init: RawInput Devices registered.
Log: [GL] Added 65536
Log: [GL] Added 131072
Init: GL_VENDOR     : NVIDIA Corporation
Init: GL_RENDERER   : GeForce 940MX/PCIe/SSE2
Init: GL_VERSION    : 4.6.0 NVIDIA 456.71
Init: Device supports: GL_EXT_bgra
Init: Device supports: GL_ARB_texture_compression
Init: Device supports: GL_EXT_texture_compression_s3tc
Init: Device supports: GL_ARB_texture_compression_rgtc
Init: Device supports: GL_ARB_texture_compression_bptc
Init: Device supports: GL_EXT_texture_shared_exponent
Init: Device supports: GL_EXT_texture_filter_anisotropic
Init: Device supports: GL_ARB_texture_filter_anisotropic
Init: Device supports: GL_NV_texture_env_combine4
Init: Device supports: GL_EXT_texture_lod_bias
Init: Device supports: GL_EXT_secondary_color
Init: Device supports: GL_ARB_multitexture
Init: Device supports: GL_EXT_multi_draw_arrays
Init: Device supports: GL_ARB_vertex_program
Init: Device supports: GL_ARB_fragment_program
Init: Device supports: GL_ARB_framebuffer_object
Init: Device supports: GL_ARB_sampler_objects
Init: Depth bits: 24
Init: AA samples: 4
Log: 4 Texture Mapping Units found
Init: MaxAnisotropy: 16
Log: BufferActorTris: 1
Log: UseDetailAlpha: 1
Log: Bound to Galaxy.dll
Init: Galaxy is using DirectSound
Init: Galaxy initialized
DevAudio: Galaxy SetViewport: WindowsViewport0
Init: Game engine

**Comment by Dr-Flay** [comment]
If this is an issue with Xpickups then you can try other maps from the map pack or try the mutator on its own
https://ut99.org/viewtopic.php?f=3&t=5082

**Comment by TankBeef** [comment]
> 
> 
> If this is an issue with Xpickups then you can try other maps from the map pack or try the mutator on its own
> https://ut99.org/viewtopic.php?f=3&t=5082

Almost every Xpickups map works fine. Most FNB maps too. It has to be something specific to that map.
I should mention, I did make the switch to XC_Engine 25 for testing and the freezing stopped. So I thought I would use it from now on, but now I get the gridcollision error on many other maps. So I switched back. Maybe something got fixed on XC_Engine that wasn't on 469? So meanwhile as a temporary fix, I have to modify the configuration file before loading the game, to be able to play certain maps.
